### PR TITLE
add pyodide-lock 0.1.0a3

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

References:
- blocks https://github.com/conda-forge/pyodide-build-feedstock/pull/6

Notes:
- upstream does not yet have a non-`a`/`b` release

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge/help-python @conda-forge/staged-recipes ready for review :lock: :heart:

File: recipes/pyodide-lock/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
-{% set version = "0.1.0a1" %}
+{% set version = "0.1.0a3" %}
```
Comment (ocefpaf) [inline recipes/pyodide-lock/meta.yaml:1]: Will they release a stable version soon? Should we apply CFEP-05 here?

https://github.com/conda-forge/cfep/blob/main/cfep-05.md

Comment (bollwyvl) [reply recipes/pyodide-lock/meta.yaml:1]: I don't believe there is a concrete upstream timeline, just following the threads on https://github.com/conda-forge/pyodide-build-feedstock/pull/6 ...

Comment (ocefpaf) [reply recipes/pyodide-lock/meta.yaml:1]: We have room for exceptions. Do you think this is one?

Comment (bollwyvl) [reply recipes/pyodide-lock/meta.yaml:1]: Sure, we ended up doing so for e.g. [lsprotocol](https://pypi.org/project/lsprotocol/) as it has a lot of production downstreams. 

As I'm doing some work upstream on this, I am pretty confident I can keep it working here, but totally understand if it's not worth the exception... while non-zero, the numbers for the downstream are still [quite low](https://anaconda.org/conda-forge/pyodide-build/files), and we could leave this open and revisit if someone complains.

**Comment by bollwyvl** [comment]
Thank!

**Pull request merged**